From f6bd61a0a81ea1dc00147dc9f878353a4ab913ea Mon Sep 17 00:00:00 2001
From: Roman Arutyunyan <arut@qip.ru>
Date: Tue, 25 Sep 2012 14:40:38 +0400
Subject: [PATCH] removed profile check when reading extra PPS bytes

---
 libavcodec/h264_ps.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavcodec/h264_ps.c b/libavcodec/h264_ps.c
index cdc6b07..9a38132 100644
--- a/libavcodec/h264_ps.c
+++ b/libavcodec/h264_ps.c
@@ -533,6 +533,8 @@ static int more_rbsp_data_in_pps(H264Context *h, PPS *pps)
     const SPS *sps = h->sps_buffers[pps->sps_id];
     int profile_idc = sps->profile_idc;
 
+    return 1;
+
     if ((profile_idc == 66 || profile_idc == 77 ||
          profile_idc == 88) && (sps->constraint_set_flags & 7)) {
         av_log(h->s.avctx, AV_LOG_VERBOSE,
-- 
1.7.10.4

